packages/advanced-logic/specs/payment-network-erc777-stream-0.1.0.md

```diff
@@ -0,0 +1,445 @@
+| **parameters.txHash**  | String | The transaction hash for documentation and metadata           | Optional      |
+| **parameters.network** | String | The network of the transaction for documentation and metadata | Optional      |
+
+### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc777-fee-superfluid" exists
+- The signer is the `payee`
+
+### warnings
+
+None.
+
+### Results
+
+An event is added to the extension state events array:
+
+|  Property             |  Value                                   |
+| --------------------- | ---------------------------------------- |
+| **name**              | Constant value: "declareReceivedPayment" |
+| **parameters**        |                                          |
+| **parameters.amount** | `amount` from parameters                 |
+| **parameters.note**   | `note` from parameters                   |
+| **parameters.txHash** | `txHash` from parameters or undefined    |
+
+## Action: declareReceivedRefund TODO : assign to request or to suite?
+
+### Parameters
+
+|                        | Type   | Description                                                   | Requirement   |
+| ---------------------- | ------ | ------------------------------------------------------------- | ------------- |
+| **id**                 | String | Constant value: "pn-erc777-fee-superfluid"                    | **Mandatory** |
+| **action**             | String | Constant value: "declareReceivedRefund"                       | **Mandatory** |
+| **parameters**         | Object |                                                               |               |
+| **parameters.amount**  | Amount | The amount declared as received, in request currency          | **Mandatory** |
+| **parameters.note**    | String | Additional information about the payment                      | Optional      |
+| **parameters.txHash**  | String | The transaction hash for documentation and metadata           | Optional      |
+| **parameters.network** | String | The network of the transaction for documentation and metadata | Optional      |
+
+### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc777-fee-superfluid" exists
+- The signer is the `payee`
+
+### warnings
+
+None.
+
+### Results
+
+An event is added to the extension state events array:
+
+|  Property              |  Value                                  |
+| ---------------------- | --------------------------------------- |
+| **name**               | Constant value: "declareReceivedRefund" |
+| **parameters**         |                                         |
+| **parameters.amount**  | `amount` from parameters                |
+| **parameters.note**    | `note` from parameters                  |
+| **parameters.txHash**  | `txHash` from parameters or undefined   |
+| **parameters.network** | `network` from parameters or undefined  |
+
+---
+
+## Interpretation
```

Comment:
CR hint: brace yourself, this is the main part where I hope you can challenge the approach

Reply:
Most of the times just checking the previous request in the series should be enough. I will challenge that.